cfme/tests/infrastructure/test_providers.py

```diff
@@ -249,63 +250,142 @@ def test_provider_crud(provider):
 class TestProvidersRESTAPI(object):
     @pytest.yield_fixture(scope="function")
     def custom_attributes(self, rest_api, setup_a_provider):
-        provider = rest_api.collections.providers[0]
-        attrs = provider.custom_attributes.action.add(
-            {"name": "ca_name1", "value": "ca_value1"}, {"name": "ca_name2", "value": "ca_value2"})
-
-        yield attrs
+        provider = rest_api.collections.providers.get(name=setup_a_provider.name)
```

Comment:
It's not the provider object, it's a the /api/providers/:id resource that corresponds to the provider object returned by `setup_a_provider` fixture.